packages/components/slot/src/Slot.tsx

```diff
@@ -8,3 +8,19 @@ export type SlotProps = PropsWithChildren<React.HTMLAttributes<HTMLElement>>
 export const Slot = forwardRef<HTMLElement, SlotProps>((props, ref) => {
   return <SlotPrimitive ref={ref} {...props} />
 })
+
+/**
+ * When using Radix `Slot` component, it will consider its first child to merge its props with.
+ * In some cases, you might need to wrap the top child with additional markup without breaking this behaviour.
+ */
+export const wrapPolymorphicSlot = (
+  asChild: boolean | undefined,
```

Comment:
```suggestion
  asChild?: boolean,
```

Reply:
Semantically it is not the same. Inside of a function it means the first parameter is optional but the next one it not. I want it to be mandatory but allow `undefined` (which is different)
![Capture d’écran 2023-06-08 à 16 31 16](https://github.com/adevinta/spark/assets/2033710/ae24f312-1bb1-4725-8b73-9f7bbcd1e5ef)

Reply:
💅, when a function takes more than two arguments, it may be more legible to use an object as the argument instead